README.md

```diff
@@ -1,4 +1,371 @@
+
+After autorest.az has done some CLI specific logic, it will render the code model into code template finally output the generated code.   
+
+
+# How to use the CLI Code Generator
+
+Users can use autorest.az to generate the Azure CLI code both in rest api specs PR pipeline and in their local environment. Guidance for generate the Azure CLI code in rest api specs PR pipeline can be found in [rest api specs documentation](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/code-gen/configure-cli.md). Here we only focus on the guidance to generate the CLI code in local.  
+
+More detailed generation guidance can be found in the [how to generate doc](https://github.com/Azure/autorest.az/blob/master/doc/how-to-generate.md)
+
+## Preparing Environment
+In order to generate the Azure CLI, we need 
+1. Nodejs v10+  
+better to be v12+
+2. Python3  
+we need a Python venv  
+```
+python -m venv env
+source ./env/bin/activate // or .\env\Scripts\Activate.ps1 in windows
+pip install azure-cli // This is must to have if for simple try out
+pip install azdev // this is optional if for simple try out. 
+```
+3. Autorest  
+it can be installed by running `npm install -g autorest@latest` 
+4. Repos (Optional)  
+if users only want to try out the generated CLI extensions, they can also build the wheel artifact by themselves with `python setup.py sdist bdist_wheel` in the azure-cli-extensions/src/{serviceName} folder.  
+* Users need to prepare their swagger and prepare  
+[azure-cli-extensions](https://github.com/Azure/azure-cli-extensions) repo if targeting at Azure CLI extensions development. Or the generated code needs some manual customization after trying out.   
+[azure-cli](https://github.com/Azure/azure-cli) repo if targeting at Azure CLI main repo modules development.   
+
+
+## Authoring readme files
+Because the Autorest.Az depends on Autorest.Clicommon and Autorest.Python, it needs three readme files the readme.az.md, readme.cli.md and readme.python.md to be ready before we use it.
+
+Users can refer to [this document](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/code-gen/configure-cli.md) for more details. 
+
+## Generating the code
+1. A simple command for authoring Azure CLI extensions would be  
+`autorest --az --use=@autorest/az@latest <path-to-the-swagger-readme.md> --azure-cli-extension-folder=<path-to-the-azure-cli-extension-repo>`
+
+2. A simple command for authoring Azure CLI main modules would be  
+`autorest --az --use=@autorest/az@latest <path-to-the-swagger-readme.md> --azure-cli-folder=<path-to-the-azure-cli-repo>`
+
+The `--use=@autorest/az@latest` will use the latest `@autorest/az` public releases, private releases can be found in [github releases](https://github.com/Azure/autorest.az/releases) if users want to try out some private releases for preview some new features and fixes.  
```

Comment:
could we also add sample for use private build package with the download url?

Reply:
added. Thanks